server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -560,10 +561,10 @@ private void startTrace(ThreadContext threadContext, RestChannel channel, String
         final Map<String, Object> attributes = Maps.newMapWithExpectedSize(req.getHeaders().size() + 3);
         req.getHeaders().forEach((key, values) -> {
             final String lowerKey = key.toLowerCase(Locale.ROOT).replace('-', '_');
-            attributes.put("http.request.headers." + lowerKey, values.size() == 1 ? values.get(0) : String.join("; ", values));
+            attributes.put("http.request.headers." + lowerKey, values == null ? "" : String.join("; ", values));
```

Comment:
How did we get a null in the first place? If the header was passed but empty wouldn't it be an empty string?

Reply:
I think there is a possibility for the map to contain a key mapping to a null list; we wrap headers from netty HttpHeaders in `Netty4HttpRequest` like this:
```
// Map.get() implementation
public List<String> get(Object key) {
    return key instanceof String ? httpHeaders.getAll((String) key) : null;
}
```
It's probably remote, but I decided to err on the safe side.

(To be precise, we did _not_ get a null here, the null was in the http.method attribute, but I wanted to be sure this was not possible elsewhere too)